editoast/src/views/v2/path/pathfinding.rs

```diff
@@ -394,9 +342,9 @@ async fn extract_location_from_path_items(
 async fn check_tracks_from_path_items(
     conn: &mut PgConnection,
     infra_id: i64,
-    track_offsets: Vec<Vec<TrackOffset>>,
+    track_offsets: &[Vec<TrackOffset>],
 ) -> Result<std::result::Result<(), PathfindingResult>> {
-    for tracks in track_offsets.clone() {
+    for tracks in track_offsets.iter() {
```

Comment:
```suggestion
    for tracks in &track_offsets {
```

Some of the syntactic sugar (aka magic) provided for `for-loops`. The `for` is desugared to something like

```rust
for tracks in IntoIterator::into_iter(track_offsets)
```

which will own the `track_offsets` and therefore yield `Vec<TrackOffset>` but if you give it a reference, `IntoIterator::into_iter(&track_offsets)` will start yielding reference on each item `&Vec<TrackOffset>`.